cvmfs/sync_union_tarball.cc

```diff
@@ -158,6 +167,21 @@ void SyncUnionTarball::Traverse() {
       }
 
       case ARCHIVE_OK: {
+        if (first_iteration && create_catalog_on_root_) {
+          std::string root =
+              GetRoot(std::string(archive_entry_pathname(entry)));
+          root = root + "/.cvmfscatalog";
+          struct archive_entry *catalog = archive_entry_clone(entry);
```

Comment:
Can we use `archive_entry_new` instead? I think we cannot tell for sure which entry we get first from the tar file, so it is not necessarily the root directory of the archive. This could lead to unexpected results if we clone a random entry.  We would then also not need `GetRoot()`.

Reply:
We can for sure un `archive_entry_new`.
The reason why I went with cloning is to already have most fields set in the entry itself. Stuff like UID and GID, possibly timestamp, etc
I believe that those fields are not so necessary for the `.cvmfscatalog` file, am I correct?

I need to think how to get the `root` where to position the catalog.
Given this command:
```
cvmfs_server ingest --tarball baz.tar --base-dir foo/ foo.cern.ch
```
It will expand into `/cvms/foo.cern.ch/foo/baz/....content_of_baz...

In this case we want the catalog to be in `/cvms/foo.cern.ch/foo/baz/`, not in  `/cvms/foo.cern.ch/foo/baz/`

Reply:
Nope, I am wrong.

The ingest works in a different way.

```
cvmfs_server ingest --tarball baz.tar --base-dir foo/ foo.cern.ch
```

It will expand into `It will expand into /cvms/foo.cern.ch/foo/....content_of_baz...`

So yes, I believe we can do without the `GetRoot`